### PR TITLE
Cpp tests

### PR DESCRIPTION
**Pull request by royvegard**
Make the test better by actually comparing the calculations with expected values. I've set the error threshold to 0.1 for most of the parameters. I'm sure you are infinitely more qualified than me to say what would be appropriate values for the thresholds.

### CODE REVIEW

**Comment by thatcadguy** [comment]
Somewhat related to this, compilation on Clang gives a result of 1120-something for dPdD2 (instead of 1118-something), but it's escaping me as to why the math would be performed that differently between compilers. If someone knows, should add the appropriate compilation flags to CMakeLists.txt for Clang to make the math match.

Would've opened this as a separate issue, but not seeing the Issues menu item at all for this repository.

**Comment by ianhbell** [comment]
Yeah, issues were disabled.  But I just turned them back on.

**Comment by royvegard** [comment]
Do you think you will be merging this?

**Comment by ianhbell** [comment]
Yeah, sorry about the delay....

How about leaving the original file and then _also_ adding the test?  Otherwise looks good.  While I would in general recommend a testing system (I use Catch: https://github.com/catchorg/Catch2), it's overkill for this problem

**Comment by royvegard** [comment]
Something like this?

File: AGA8CODE/C/Detail.cpp [hunk 1]
```diff
@@ -1,5 +1,3 @@
-
```
Comment (ianhbell) [inline AGA8CODE/C/Detail.cpp]: Can you revert this whitespace-only change?

Comment (royvegard) [reply AGA8CODE/C/Detail.cpp]: Done.

**Comment by ianhbell** [comment]
Thanks!

**Pull request merged**